### PR TITLE
BITAU-69 Check for OS version in AuthenticatorBridgeManager

### PR DESCRIPTION
**Pull request by ahaisting-livefront**
## 🎟️ Tracking

https://livefront.atlassian.net/browse/BITAU-69

## 📔 Objective

The goal of this PR is to add a check for OS version to the bridge coroutines wrapper. 

## 📸 Screenshots

N/A

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt [hunk 1]
```diff
@@ -58,7 +60,9 @@ internal class AuthenticatorBridgeManagerImpl(
      */
     private val mutableSharedAccountsStateFlow: MutableStateFlow<AccountSyncState> =
         MutableStateFlow(
-            if (isBitwardenAppInstalled()) {
+            if (isBuildVersionBelow(Build.VERSION_CODES.S)) {
```
Comment (brian-livefront) [inline authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt:63]: Can we split this up into a `when` now?

File: authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt [hunk 2]
```diff
@@ -102,6 +106,10 @@ internal class AuthenticatorBridgeManagerImpl(
     }
 
     private fun bindService() {
+        if (isBuildVersionBelow(Build.VERSION_CODES.S)) {
+            mutableSharedAccountsStateFlow.value = AccountSyncState.OSVersionNotSupported
+            return
+        }
```
Comment (brian-livefront) [inline authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt:112]: 👍

File: authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/model/AccountSyncState.kt [hunk 3]
```diff
@@ -27,6 +27,11 @@ sealed class AccountSyncState {
      */
     data object Loading : AccountSyncState()
 
+    /**
+     * OS version can't support account syncing.
+     */
+    data object OSVersionNotSupported: AccountSyncState()
```
Comment (brian-livefront) [inline authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/model/AccountSyncState.kt:33]:
```suggestion
    data object OsVersionNotSupported: AccountSyncState()
```

File: authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt (hunk 1 above)
Comment (ahaisting-livefront) [reply authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt]: Ah, yep, that's fair!

b97e8f12560f9413f57e2415d53f9301338a2d94

File: authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/model/AccountSyncState.kt (hunk 3 above)
Comment (ahaisting-livefront) [reply authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/model/AccountSyncState.kt]: 4411c011b76223c4a8725e53bed6ead87f58c98b

File: authenticatorbridge/src/test/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerTest.kt [hunk 4]
```diff
@@ -87,6 +93,26 @@ class AuthenticatorBridgeManagerTest {
...
+            connectionType = AuthenticatorBridgeConnectionType.DEV,
+            symmetricKeyStorageProvider = fakeSymmetricKeyStorageProvider,
+            callbackProvider = testAuthenticatorBridgeCallbackProvider,
+            processLifecycleOwner = fakeLifecycleOwner,
+        )
+        assertEquals(AccountSyncState.OsVersionNotSupported, manager.accountSyncStateFlow.value)
+    }
+
+    @Test
+    fun `onStart when OS level is below S should set state to OsNotSupported`() {
```
Comment (brian-livefront) [inline authenticatorbridge/src/test/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerTest.kt:110]:
```suggestion
    fun `onStart when OS level is below S should set state to OsVersionNotSupported`() {
```

Comment (ahaisting-livefront) [reply authenticatorbridge/src/test/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerTest.kt]: Thank you. 

596b616e4af3a52754358d3262b4544e37fbd023

**Review by brian-livefront — APPROVED** [review]

**Comment by ahaisting-livefront** [comment]
Thanks, @brian-livefront !

**Pull request merged**